src/write_gcm.c

```diff
@@ -3244,12 +3274,18 @@ static int wg_json_CreateTimeSeries(
     wg_json_string(jc, "metric");
     wg_json_Metric(jc, head);
 
+    char *metric_value_type = wg_metric_value_type_from_wg_payload_t(head);
+
     switch (head->values[0].ds_type) {
       case DS_TYPE_GAUGE:
       wg_json_string(jc, "metricKind");
       wg_json_string(jc, "GAUGE");
       wg_json_string(jc, "valueType");
-      wg_json_string(jc, "DOUBLE");
+      if (metric_value_type != NULL) {
```

Comment:
I'm kind of on the fence about this one. We could either leave these conditionals everywhere, or embed the logic for figuring out the default type into `wg_metric_value_type_from_wg_payload`, so it defaults gauges to `"DOUBLE"` and counters/derives to `"INT64"`. Up to you.